### PR TITLE
feat: support reverse proxy header config

### PR DESCRIPTION
**Pull request by jawnsy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ View [our docs](https://coder.com/docs/setup/installation) for detailed installa
...
...ame","operator":"In","values":["coderd"]}]},"topologyKey":"kubernetes.io/hostname"},"weight":1}]}},"builtinProviderServiceAccount":{"annotations":{},"labels":{}},"devurlsHost":"","extraLabels":{},"image":"","networkPolicy":{"enable":true},"oidc":{"enableRefresh":false,"redirectOptions":{}},"podSecurityContext":{"runAsGroup":1000,"runAsNonRoot":true,"runAsUser":1000,"seccompProfile":{"type":"RuntimeDefault"}},"proxy":{"exempt":"cluster.local","http":"","https":""},"replicas":1,"resources":{"limits":{"cpu":"250m","memory":"512Mi"},"requests":{"cpu":"250m","memory":"512Mi"}},"reverseProxy":{"headers":[],"trustedOrigins":[]},"satellite":{"accessURL":"","enable":false,"primaryURL":""},"securityContext":{"allowPrivilegeEscalation":false,"readOnlyRootFilesystem":true,"runAsGroup":1000,"runAsNonRoot":true,"runAsUser":1000,"seccompProfile":{"type":"RuntimeDefault"}},"serviceAnnotations":{},"serviceNodePorts":{"http":null,"https":null},"serviceSpec":{"externalTrafficPolicy":"Local","loadBalancerIP":"","loadBalancerSourceRanges":[],"type":"LoadBalancer"},"superAdmin":{"passwordSecret":{"key":"password","name":""}},"tls":{"devurlsHostSecretName":"","hostSecretName":""},"trustProxyIP":false}` |
```
Comment (bryphe-coder) [inline README.md:28]: Looks like just the `reverseProxy` config is added here - seems good to me

File: README.md [hunk 2]
```diff
@@ -71,7 +74,7 @@ View [our docs](https://coder.com/docs/setup/installation) for detailed installa
 | coderd.tls | object | TLS configuration for coderd. These options will override dashboard configuration. | `{"devurlsHostSecretName":"","hostSecretName":""}` |
 | coderd.tls.devurlsHostSecretName | string | The secret to use for DevURL TLS. | `""` |
 | coderd.tls.hostSecretName | string | The secret to use for TLS. | `""` |
-| coderd.trustProxyIP | bool | Whether Coder should trust X-Real-IP and/or X-Forwarded-For headers from your reverse proxy. This should only be turned on if you're using a reverse proxy that sets both of these headers. This is always enabled if the Nginx ingress is deployed. | `false` |
+| coderd.trustProxyIP | bool | Configures Coder to accept X-Real-IP and X-Forwarded-For headers from any origin. This option is deprecated and will be removed in a future release. Use the coderd.reverseProxy setting instead, which supports configuring an allowlist of trusted origins. | `false` |
```
Comment (bryphe-coder) [inline README.md:77]: Thanks for updating this documentation re: deprecation (and suggesting a migration path)

File: templates/coderd.yaml [hunk 3]
```diff
@@ -183,6 +181,12 @@ spec:
               value: {{ .Values.coderd.proxy.https | quote }}
             - name: no_proxy
               value: {{ .Values.coderd.proxy.exempt | quote }}
+            - name: TRUST_PROXY_IP
+              value: {{ .Values.coderd.trustProxyIP | quote }}
+            - name: PROXY_TRUSTED_ORIGINS
+              value: {{ join "," .Values.coderd.reverseProxy.trustedOrigins | quote }}
+            - name: PROXY_TRUSTED_HEADERS
+              value: {{ join "," .Values.coderd.reverseProxy.headers | quote }}
```
Comment (bryphe-coder) [inline templates/coderd.yaml:189]: Looks reasonable to me - but I'm not a helm expert. Glad you have the test cases, too!

**Review by bryphe-coder — APPROVED** [review]

**Comment by bryphe-coder** [comment]
I'm not a helm expert, but everything seems OK to me 👍

File: README.md (hunk 2 above)
Comment (jawnsy) [reply README.md:77]: the docs are generated from the comments in the values file, btw :wink:

**Pull request merged**